gravitee-apim-repository/gravitee-apim-repository-elasticsearch/src/main/java/io/gravitee/repository/elasticsearch/v4/log/adapter/connection/SearchConnectionLogResponseAdapter.java

```diff
@@ -38,11 +38,30 @@ public static LogResponse<ConnectionLog> adapt(SearchResponse response) {
 
         return new LogResponse<>(
             (int) hits.getTotal().getValue(),
-            hits.getHits().stream().map(h -> buildFromSource(h.getSource())).toList()
+            hits.getHits().stream().map(h -> buildFromSource(h.getIndex(), h.getId(), h.getSource())).toList()
         );
     }
 
-    private static ConnectionLog buildFromSource(JsonNode json) {
+    private static ConnectionLog buildFromSource(String index, String id, JsonNode json) {
+        if (index.contains("request")) {
+            return ConnectionLog
+                .builder()
+                .requestId(id)
+                .timestamp(asTextOrNull(json.get("@timestamp")))
+                .applicationId(asTextOrNull(json.get("application")))
+                .apiId(asTextOrNull(json.get("api")))
+                .planId(asTextOrNull(json.get("plan")))
+                .clientIdentifier(asTextOrNull(json.get("subscription")))
+                .transactionId(asTextOrNull(json.get("transaction")))
+                .method(HttpMethod.get(asIntOr(json.get("method"), 0)))
+                .status(asIntOr(json.get("status"), 0))
+                .requestEnded(true)
+                .entrypointId(null)
+                .gateway(asTextOrNull(json.get("gateway")))
+                .uri(asTextOrNull(json.get("uri")))
+                .gatewayResponseTime(json.get("response-time").asLong(0L))
+                .build();
+        }
```

Comment:
What about creating a builder with common fields between V2 and V4 indexes. And only add specific fields in the `if` statement?